server/src/main/java/org/elasticsearch/repositories/RepositoriesService.java

```diff
@@ -429,9 +429,7 @@ public void applyClusterState(ClusterChangedEvent event) {
                 Repository repository = survivors.get(repositoryMetadata.name());
                 if (repository != null) {
                     // Found previous version of this repository
-                    RepositoryMetadata previousMetadata = repository.getMetadata();
-                    if (previousMetadata.type().equals(repositoryMetadata.type()) == false
-                            || previousMetadata.settings().equals(repositoryMetadata.settings()) == false) {
```

Comment:
All the other repositories are recreated when the settings change.
But the encrypted repo purposefully accepts that some settings (all the password related ones) can be changed on the fly, and is prepared to deal with it.